### ISSUE TITLE
Postgres: RF03 false positive with keyword arguments

### ISSUE THREAD

**Issue by fayyazul-centaurlabs** [sqlfluff/sqlfluff#6290]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Passing an argument to a function as a keyword argument triggers RF03. It looks like it's the same issue as #6189 but for postgres.

### Expected Behaviour

Function parameters should not be qualified.

### Observed Behaviour
see "How to Reproduce".

### How to reproduce

The error is raised on the parameter `a` in the call below.
```sql
drop function if exists __test__(a integer);
create or replace function __test__(a integer)
returns table (b int, c int)
language plpgsql
as $$
begin
    return query
    select a, a;
end
$$;

select t1.b
from __test__(a := 1) t1;
```
```shell
$ sql lint test.sql
== [test.sql] FAIL                                                                                   
L:  13 | P:  15 | RF03 | Unqualified reference 'a' found in single table select.
                       | [references.consistent]
L:  13 | P:  15 | RF03 | Unqualified reference 'a' found in single table select
                       | which is inconsistent with previous references.
                       | [references.consistent]
All Finished 📜 🎉!
```

Running fix on the code changes it to
```sql
select t1.b
from __test__(t1.a := 1) t1;
```
which is invalid code.

Note that this error is not raised if using the other syntax for keyword arguments `=>`.
The following does not raise an error.
```sql
select t1.b
from __test__(a => 1) t1;
```

### Dialect

Postgres

### Version

sqlfluff, version 3.2.0
Python 3.12.2

### Configuration
```
[sqlfluff]
dialect = postgres
sql_file_exts = .sql

exclude_rules =
    AL01,
    AL02,
    AM04

[sqlfluff:rules:layout.commas]

[sqlfluff:layout:type:comma]
line_position = leading
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
